src/array/cluster.ts

```diff
@@ -13,6 +15,14 @@ export function cluster<T, Size extends number = 2>(
   array: readonly T[],
   size: Size = 2 as Size,
 ): Cluster<T, Size>[] {
+  if (size <= 0 || Number.isNaN(size)) {
+    throw new IllegalSizeError(`Size must be 1 or more, the size is ${size}`)
```

Comment:
I think returning `[]` when `size <= 0` would be better. It's more concise, makes logical sense, and works fine with the `Cluster<T, Size>[]` return type.